runtime/lua/vim/_editor.lua

```diff
@@ -917,6 +917,31 @@ function vim._expand_pat(pat, env)
 
   local final_env = env
 
+  --- @param m table
+  --- @param k string
+  --- @return any
+  local function safe_tbl_get(m, k)
+    local val = rawget(m, k)
+    if val ~= nil then
+      return val
+    end
+
+    local mt = getmetatable(m)
+    if not mt then
+      return m == vim and vim._extra[k] or nil
+    end
+
+    -- use mt.__index, _submodules as fallback
+    if type(mt.__index) == 'table' then
+      return rawget(mt.__index, k)
+    end
+
+    local sub = rawget(m, '_submodules')
+    if sub and type(sub) == 'table' and rawget(sub, k) then
+      return m[k] -- access to required module
```

Comment:
Just noticed this in the commit message:

> Always access `._submodules` field to ensure `_defer_require` module loaded.

That is a rather important thing to clearly call out. Starting with this code comment. Is this comment correct?:


```suggestion
      -- Access the module to force _defer_require() to load the module.
      return m[k]
```

Reply:
Yes. I should elaborate on it..